DashAI/back/preprocessor/base_preprocessor.py

```diff
@@ -0,0 +1,24 @@
+from abc import abstractmethod
+from typing import Final
+
+
+class BasePreprocessor:
+    """Base class for DashAI preprocessor."""
+
+    TYPE: Final[str] = "Preprocessor"
+
+    @abstractmethod
+    def process(self, dataset):
```

Comment:
I trying to enfoce that every function and method use a typing notation (although is not always possible).
Here, it could be (I don't know exactly, so check it in runtime for example (with `type`)).

We are also trying to use scikitlearn's conventions for naming methods. That is, fit to fit (whatever that is), transform for any data transformation, and predict to apply the predictor.

```suggestion
    def transform(self, dataset: DashAIDataset) -> DashAIDataset:
```

We also